src/main/java/org/springframework/data/elasticsearch/core/SearchHit.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core;
+
+import org.springframework.util.Assert;
+
+/**
+ * Encapsulates the found data with additional information from the search.
+ * 
+ * @param <T> the result data class.
+ * @author Peter-Josef Meisch
+ * @since 4.0
+ */
+public final class SearchHit<T> {
+
+	private final float score;
+	private final T hit;
+
+	public static <T> Builder<T> builder() {
```

Comment:
Let's avoid the builder pattern for simple objects such as this one. Having a public constructor is fine for a simple domain object. All of our [domain objects](https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/geo/GeoPage.java) in commons follow this idea which makes it easy to use and reuse such types.